lib/shared/store.js

```diff
@@ -198,24 +194,35 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
           !constraints.isSystemDrive(drive)
 
         ])) {
+          // Auto-select this drive
           return storeReducer(newState, {
             type: ACTIONS.SELECT_DRIVE,
             data: drive.device
           })
         }
-      }
 
-      const selectedDevice = newState.getIn([ 'selection', 'drive' ])
-
-      if (selectedDevice && !_.find(action.data, {
-        device: selectedDevice
-      })) {
+        // Deselect this drive in case it still is selected
         return storeReducer(newState, {
-          type: ACTIONS.DESELECT_DRIVE
+          type: ACTIONS.DESELECT_DRIVE,
+          data: drive.device
         })
       }
 
-      return newState
+      const selectedDevices = newState.getIn([ 'selection', 'devices' ]).toJS()
+
+      return _.reduce(selectedDevices, (accState, device) => {
```

Comment:
Could you add a code-comment explaining what this is doing?
I _think_ it's deselecting all drives? Would it be easier to just have a `ACTIONS.DESELECT_ALL_DRIVES` ? (which you could do just once _before_ testing the number of drives, so that you don't have two separate code-paths (as you have now) each doing `ACTIONS.DESELECT_DRIVE`)

Reply:
It's deselecting drives that don't exist in `availableDrives` anymore – I've added a few comments explaining that block now.